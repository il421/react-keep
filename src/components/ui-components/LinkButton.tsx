import React from "react";
import "../../styles/ui-components/_link-button.scss";

export interface LinkButtonProps
  extends Partial<Pick<HTMLButtonElement, "disabled" | "className" | "id">> {
  text: string;
  type: "submit" | "button";
  onClick?: (data: any) => void;
}

/**
 * The Link button just looks like a styled link, in fact it is just a button
 */
export class LinkButton extends React.PureComponent<LinkButtonProps> {
  render() {
    return (
      <button
        id={this.props.id}
        className={`link-button ${this.props.className}`}
        type={this.props.type}
        disabled={this.props.disabled}
        onClick={this.props.onClick}
      >
        {this.props.text}
      </button>
    );
  }
}
