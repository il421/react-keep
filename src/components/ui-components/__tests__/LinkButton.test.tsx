import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import { LinkButton, LinkButtonProps } from "../LinkButton";

let props: LinkButtonProps, wrapper: ShallowWrapper;

beforeEach(() => {
  props = {
    text: "text",
    type: "submit"
  };
});

test("should render LinkButton", () => {
  wrapper = shallow(<LinkButton {...props} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
