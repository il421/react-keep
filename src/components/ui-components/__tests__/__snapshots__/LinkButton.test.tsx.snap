// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LinkButton 1`] = `
<button
  className="link-button undefined"
  type="submit"
>
  text
</button>
`;
